src/arch/src/aarch64/gic/regs.rs

```diff
@@ -1,24 +1,45 @@
 // Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-mod dist_regs;
-mod icc_regs;
-mod redist_regs;
-
-use crate::aarch64::gic::{Error, Result};
-use kvm_bindings::kvm_device_attr;
-use kvm_ioctls::DeviceFd;
 use std::fmt::Debug;
 use std::iter::StepBy;
 use std::ops::Range;
+
+use kvm_bindings::kvm_device_attr;
+use kvm_ioctls::DeviceFd;
 use versionize::{VersionMap, Versionize, VersionizeResult};
 use versionize_derive::Versionize;
 
+use crate::aarch64::gic::{Error, Result};
+
 #[derive(Debug)]
 pub struct GicRegState<T: Versionize> {
     pub(crate) chunks: Vec<T>,
 }
 
+/// Structure for serializing the state of the Vgic ICC regs
+#[derive(Debug, Default, Versionize)]
+pub struct VgicSysRegsState {
```

Comment:
I feel like this structure is GIC-version dependent and that it was created for GICv3 and then shoehorned for V2 as well.